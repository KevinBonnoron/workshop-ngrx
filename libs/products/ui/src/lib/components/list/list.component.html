<h1>Liste des produits</h1>

<ng-container *ngIf="products$ | async as products">
  <div class="grid grid-cols-4 gap-4">
    <products-ui-card *ngFor="let product of products" [product]="product"></products-ui-card>
  </div>
  
  <div class="pt-4">
    <button mat-raised-button color="primary" [routerLink]="['/add']">Ajouter un produit</button>
  </div>
</ng-container>
