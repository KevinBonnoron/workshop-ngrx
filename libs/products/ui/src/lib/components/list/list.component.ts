import { AsyncPipe, NgFor, NgIf } from '@angular/common';
import { Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { RouterLink } from '@angular/router';
import { ProductsService } from '@workshop/products-data-access';
import { CardComponent } from '../card/card.component';

@Component({
  standalone: true,
  imports: [MatButtonModule, CardComponent, NgFor, NgIf, RouterLink, AsyncPipe],
  selector: 'products-ui-list',
  templateUrl: './list.component.html'
})
export class ListComponent {
  private readonly productsService = inject(ProductsService);

  readonly products$ = this.productsService.findAll();
}
