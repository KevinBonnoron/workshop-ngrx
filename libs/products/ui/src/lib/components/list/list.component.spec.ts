import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ProductsService } from '@workshop/products-data-access';
import { of } from 'rxjs';
import { ListComponent } from './list.component';

describe('ListComponent', () => {
  let fixture: ComponentFixture<ListComponent>;
  let component: ListComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ListComponent],
      providers: [
        { provide: ProductsService, useValue: { findAll: jest.fn(() => of([])) } }
      ]
    });

    fixture = TestBed.createComponent(ListComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snaptshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
