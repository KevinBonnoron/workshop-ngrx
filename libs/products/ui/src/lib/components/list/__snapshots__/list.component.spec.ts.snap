// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent should match snaptshot 1`] = `
<products-ui-list
  products$={[Function Observable]}
  productsService={[Function Object]}
>
  <h1>
    Liste des produits
  </h1>
</products-ui-list>
`;
